envoy/http/header_map.h

```diff
@@ -449,6 +449,16 @@ class HeaderMap {
    */
   virtual uint64_t byteSize() const PURE;
 
+  /**
+   * @return uint32_t the max size of the header map in kilobyte.
+   */
+  virtual uint32_t maxHeadersKb() const PURE;
+
+  /**
+   * @return uint32_t the max count of headers in a header map.
+   */
+  virtual uint32_t maxHeadersCount() const PURE;
```

Comment:
A high level question: Is this `max_count` applicable to h1 and h2? 

AFAIC this is added mostly for h3, but since you already plumbed max_size into h3, the goal should be achieved? If it is not really applicable to h1 and h2, do we still need to add the header count in this top abstract layer?

Reply:
H1 check linked above doesn't check against count. But i guess it doesn't hurt to have this as the other codecs are using it. Thanks for checking/confirming!

Reply:
Here is the H1 checks for count: https://github.com/envoyproxy/envoy/blob/3e29621531e42c4f2aa8562f35b03d74db160bd7/source/common/http/http1/codec_impl.cc#L574